api/client/cli.go

```diff
@@ -25,33 +24,12 @@ import (
 // DockerCli represents the docker command line client.
 // Instances of the client can be returned from NewDockerCli.
 type DockerCli struct {
-	// initializing closure
-	init func() error
-
-	// configFile has the client configuration file
 	configFile *configfile.ConfigFile
-	// in holds the input stream and closer (io.ReadCloser) for the client.
-	in io.ReadCloser
-	// out holds the output stream (io.Writer) for the client.
-	out io.Writer
-	// err holds the error stream (io.Writer) for the client.
-	err io.Writer
-	// keyFile holds the key file as a string.
-	keyFile string
-	// inFd holds the file descriptor of the client's STDIN (if valid).
-	inFd uintptr
-	// outFd holds file descriptor of the client's STDOUT (if valid).
-	outFd uintptr
-	// isTerminalIn indicates whether the client's STDIN is a TTY
-	isTerminalIn bool
-	// isTerminalOut indicates whether the client's STDOUT is a TTY
-	isTerminalOut bool
-	// client is the http client that performs all API operations
-	client client.APIClient
-	// inState holds the terminal input state
-	inState *term.State
-	// outState holds the terminal output state
-	outState *term.State
+	in         *InStream
```

Comment:
I don't know how much value these comments provide. They are basically just rewording what the type system already tells us.